id-repository/id-repository-identity-service/src/main/java/io/mosip/idrepository/identity/service/impl/IdRepoServiceImpl.java

```diff
@@ -509,19 +507,33 @@ private void updateRequestBodyData(IdRequestDTO request) throws IdRepoAppExcepti
 	}
 
 	@SuppressWarnings({ "rawtypes", "unchecked" })
-	protected void updateVerifiedAttributes(IdRequestDTO<T> requestDTO, DocumentContext inputData, DocumentContext dbData) {
+	protected void updateVerifiedAttributes(IdRequestDTO<T> requestDTO, DocumentContext inputData, DocumentContext dbData) throws IdRepoAppException {
 		List dbVerifiedAttributes = (List) dbData.read(".verifiedAttributes");
 		dbVerifiedAttributes.remove(null);
+		boolean isV2Version = requestDTO.getVerifiedAttributes() instanceof Map ? true : false;
 		if (dbVerifiedAttributes.isEmpty()) {
-			dbVerifiedAttributes.add(new ArrayList<>());
+			dbVerifiedAttributes.add(isV2Version ? new HashMap<>() : new ArrayList<>());
 		}
-		List verifiedAttributeList = (List) dbVerifiedAttributes.get(0);
-		if (Objects.nonNull(requestDTO.getVerifiedAttributes())) {
-			verifiedAttributeList.addAll((List<String>)requestDTO.getVerifiedAttributes());
+		if (isV2Version) {
+			Map dbVerifiedAttributeMap = (Map) dbVerifiedAttributes.get(0);
+			if (Objects.nonNull(requestDTO.getVerifiedAttributes())
+					&& !((Map) requestDTO.getVerifiedAttributes()).isEmpty()) {
+				dbVerifiedAttributeMap.putAll((Map) requestDTO.getVerifiedAttributes());
+			} else {
```

Comment:
Logic to remove keys from verifiedAttributes map must be applied whether or not verifiedAttributes is non-null or null.
On update of a field, should retain the entry for the same field in verifiedAttributes only if the request has verification details otherwise we should remove the entry for that key from verifiedAttributes map.

Reply:
Changes done